src/pytest_workflow/plugin.py

```diff
@@ -129,13 +142,78 @@ def pytest_collection(session):
     print()
 
 
-def pytest_runtestloop(session):
+def pytest_collection_modifyitems(config: PytestConfig,
+                                  items: List[pytest.Item]):
+    """Here we skip all tests related to workflows that are not executed"""
+
+    for item in items:
+        marker = item.get_closest_marker(
+            name="workflow")  # type: Optional[MarkDecorator] # noqa: E501
+        if marker is not None:
+            workflow_name = marker.kwargs.get('name')
+            if workflow_name is None:
+                # If name key is not defined use the first arg.
+                if len(marker.args) >= 1:
+                    workflow_name = marker.args[0]
+                    # Make sure a name attribute is added anyway for the
+                    # fixture lookup.
+                    marker.kwargs['name'] = workflow_name
+                else:
+                    # If we raise an error here a number of things will happen:
+                    # + Pytest will crash. Giving a lot of INTERNAL ERROR lines
+                    # + No tests will run
+                    # + No workflows will be started
+                    # + All the threads that are waiting for a workflow to
+                    #   finish will wait indefinitely. Causing an infinite
+                    #   hang. Pytest will never finish.
+                    # Therefore we do not crash here, but raise a warning.
+                    item.warn(pytest.PytestWarning(
```

Comment:
What about using a `TypeError` as the warning here? That's python's exception that raised when a function is called with too few arguments.

Reply:
If the warning is not a `pytest.PytestWarning` type, then pytest will throw a tantrum and not accept it.
`item.warn` is quite useful because it will display the warn message in a summary after the tests have been completed. So it is worth it here to play by pytest's rules.

Reply:
Could've expected pytest doing non-standard things again ;-)